Added helper functions for facility xwalks

erika-tyagi opened the pull request:
Two more quick helper functions, useful in the short-term for facility crosswalks, potentially for other workflows too? 

Right now, the only valid state spellings are the 50 states, DC, Puerto Rico, Guam, Virgin Islands, and Not Available. As a result, invalid spellings (intentionally) include Federal, District of Columbia, and Abroad. Is this too strict? Is there anything else I should include as valid? 

Also, the current crosswalk includes Not Available, NA, Abroad, DC and District of Columbia listed separately, Deleware and Delaware listed separately, etc. I can clean this up when I reset the IDs - unless it would break things? 

Opening PRs that grant statehood to territories is a lot of pressure.

nmmarquez commented:
Hey I think these are useful functions, is it possible vectorize these functions so that they work with more than 1 HIFLD or state at a time?

right now I get weird warnings and variable output when passing a vector of id values or states

hjohns12 commented:
It looks like there is a prison in American Samoa called Tafuna Correctional Facility (pretty sure we're not collecting data about it though). Right now, I think this would get categorized under "Not available". To be consistent, I think we should add in the other US territories to the list of possible states. 

Also just discovered that there is a prison in the Northern Mariana Islands, although I couldn't find any COVID data online.

nmmarquez commented on R/is_valid_state.R:
```diff
@@ -0,0 +1,36 @@
+    
+    valid_states <- datasets::state.name %>% 
+        append(c(
+            "DC", 
```
I think we are changing DC to District of Columbia in the new info sheet

erika-tyagi commented:
Thanks! Wholehearted agree with all comments. Updates:  
* Vectorized the functions (and added examples of passing multiple IDs/states in) 
* Added all 5 of the populated territories 
* Changed the default `na_is_na` to F. The main reason I initially wrote them to return T/F/NA instead of T/F is because I wanted to handle cases where a user didn't list an HIFLD (NA) differently from a case where they listed an invalid HIFLD (typo) in the Google Sheet - but I'm open to removing that option from these functions! 
* Changed DC to District of Columbia

hjohns12 commented:
Looks ready to merge to me!

nmmarquez commented:
So im a big dummy and misinterpreted the `na_is_na` paramter. I totally think it should be `TRUE` after testing. @erika-tyagi can we 1) switch it back, and 2) please forgive me

erika-tyagi commented:
> So im a big dummy and misinterpreted the `na_is_na` paramter. I totally think it should be `TRUE` after testing. @erika-tyagi can we 1) switch it back, and 2) please forgive me

@nmmarquez Done! Renamed to `missing_as_na` and defaults to TRUE. You're the opposite of a dummy, and very much forgiven!
